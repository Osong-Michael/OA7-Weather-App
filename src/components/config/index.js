import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import weatherReducer from '../Redux/reducers';

export const storeFactory = (initialState) => {
  return createStore(weatherReducer, {...initialState}, applyMiddleware(thunk));
}

export const findByTestId = (wrapper, val) => {
  return wrapper.find(`[data-test-id="${val}"]`);
}