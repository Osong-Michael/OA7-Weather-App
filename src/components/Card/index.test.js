import { mount } from 'enzyme';
import CardBody from './';
import { findByTestId } from '../config';
import data from '../../test_data/data.json'


describe('Card Body Renders correctly', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(< CardBody  weather={data.list[0]} />)
  });

  test('renders the entire card block', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('shows the weather description of a particular day', () => {
    const description = findByTestId(wrapper, 'weather-description')
    expect(description.text()).toEqual('Light Rain');
  });

  test('shows the weather "Max Temperature" of a particular day', () => {
    const maxTemp = findByTestId(wrapper, 'weather-max-temp')
    expect(maxTemp.text()).toEqual('Max Temperature');
  });

  test('shows the weather "Min Temperature" of a particular day', () => {
    const minTemp = findByTestId(wrapper, 'weather-min-temp')
    expect(minTemp.text()).toEqual('Min Temperature');
  });

});