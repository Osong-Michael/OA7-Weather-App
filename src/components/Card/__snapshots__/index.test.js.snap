// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Body Renders correctly renders the entire card block 1`] = `ReactWrapper {}`;
